src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -99,6 +105,7 @@ private void initialize(NativeMemoryCacheManagerDto nativeMemoryCacheDTO) {
 
         if (nativeMemoryCacheDTO.isExpirationLimited()) {
             cacheBuilder.expireAfterAccess(nativeMemoryCacheDTO.getExpiryTimeInMin(), TimeUnit.MINUTES);
+            this.cacheMaintainer = new ScheduledExecutor(() -> cache.cleanUp(), 60 * 1000);
```

Comment:
Could you wrap `cache.cleanUp` with try-catch? 
A task throwing an exception will never be rescheduled.